src/lib.rs

```diff
@@ -823,7 +823,7 @@ impl Connection {
     /// Flush a NATS connection by sending a `PING` protocol and waiting for the responding `PONG`.
     /// Will fail with `TimedOut` if the server does not respond with in 10 seconds.
     /// Will fail with `NotConnected` if the server is not currently connected.
-    /// Will fail with `BrokenPipe` if the connection to the server is lost.
+    /// Will fail with `ConnectionReset` if the connection to the server is lost.
```

Comment:
nit but we do not know yet if it has been reset, we just know we got disconnected at this point. Other client use disconnect(ed) as an event and state type.

Reply:
we're using io::Error instead of creating our own. my opinion is that it really reduces user friction (especially for network-related issues) to just use the std error type around this. https://github.com/nats-io/nats.rs/pull/29#issuecomment-620035458

Reply:
Disconnected is not an option among those: https://doc.rust-lang.org/std/io/enum.ErrorKind.html

I don't have a preference for which one gets used, but I do think it should be one of these rather than forcing users to handle our own error type.

Reply:
I agree with using standard errors, so fair point. That is why I coded it that way and did not create our own. Thx.

Reply:
BrokenPipe is sent back or ConnectReset? Which means by peer IIRC.

Reply:
I chose BrokenPipe because that's what happens when it's not necessarily the result of a rst/fin. ConnectionReset is a rst. Which would you prefer from https://doc.rust-lang.org/std/io/enum.ErrorKind.html?

Reply:
Let's go with your original one, thanks for the patience.